src/test/java/org/takes/misc/VerboseListTest.java

```diff
@@ -0,0 +1,407 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.misc;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * Tests for {@link VerboseList}.
+ *
+ * @author I. Sokolov (happy.neko@gmail.com)
+ * @version $Id$
+ */
+@SuppressWarnings("PMD.TooManyMethods")
+@RunWith(MockitoJUnitRunner.class)
+public class VerboseListTest {
+
+    /**
+     * Custom exception message.
+     */
+    private static final String MSG = "Error message";
+
+    /**
+     * Rule for asserting thrown exceptions.
+     */
+    @Rule
+    public final transient ExpectedException thrown = ExpectedException.none();
+
+    /**
+     * Decorated List.
+     */
+    @Mock
+    private transient List<Object> origin;
+
+    /**
+     * Decorator.
+     */
+    private transient VerboseList<Object> list;
+
+    /**
+     * Creates decorator.
+     * @throws Exception If something goes wrong.
+     */
+    @Before
+    public final void setUp() throws Exception {
+        this.list = new VerboseList<Object>(this.origin, MSG);
+    }
+
+    /**
+     * VerboseList should delegate size method to decorated List.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void delegatesSize() throws Exception {
+        this.list.size();
+        Mockito.verify(this.origin).size();
+    }
+
+    /**
+     * VerboseList should delegate isEmpty method to decorated List.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void delegatesIsEmpty() throws Exception {
+        this.list.isEmpty();
+        Mockito.verify(this.origin).isEmpty();
+    }
+
+    /**
+     * VerboseList should delegate contains method to decorated List.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void delegatesContains() throws Exception {
+        final Object obj = new Object();
+        this.list.contains(obj);
+        Mockito.verify(this.origin).contains(obj);
+    }
+
+    /**
+     * VerboseList should return {@link VerboseIterator}.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public final void returnsVerboseIterator() throws Exception {
+        final Iterator<Object> iterator = this.list.iterator();
```

Comment:
@Happy-Neko We can avoid inline this variable to the first argument in `assertThat` instead of declaring a variable and using it just once.